plugins/modules/postgresql_subscription.py

```diff
@@ -732,13 +739,20 @@ def main():
                                           check_mode=module.check_mode)
 
         else:
-            if connparams:
-                connparams = cast_connparams(connparams)
-
-            changed = subscription.update(connparams,
-                                          publications,
-                                          subsparams,
-                                          check_mode=module.check_mode)
+            if hasattr(subscription, 'connparams'):
+                if connparams:
+                    connparams = cast_connparams(connparams)
+
+                changed = subscription.update(connparams,
+                                              publications,
+                                              subsparams,
+                                              check_mode=module.check_mode)
+            else:
+                module.warn("'connparams' is ignored when existing subscription connparms are unknowable")
```

Comment:
This wording is a bit weird. Open to suggestions here on a different way to phrase this so that it is clear why the module won't attempt to make a change to the `connparams`

Reply:
SGTM, Let's discuss it in a separate issue, thanks for spotting this!